contracts/CandidateHub.sol

```diff
@@ -2,23 +2,23 @@
 pragma solidity 0.8.4;
 import "./lib/BytesToTypes.sol";
 import "./lib/Memory.sol";
+import {ReentrancyGuard} from "./lib/ReentrancyGuard.sol";
```

Comment:
I am aware of the fact that the relevant functions are structured as checks-effects-interactions. I'm still adding a guard as a 2nd layer protection against future (unsafe) changes